views/providers.py

```diff
@@ -185,10 +197,11 @@ def delete_hawkular_provider(self):
 
         # Verify if the provider is deleted from the provider list by refreshing the page.
 
-        self.web_driver.refresh()
+        count = 0
         while ui_utils(self.web_session).isElementPresent(By.XPATH, "//a[contains(@title,'Name: {}')]".format(
-                self.web_session.HAWKULAR_PROVIDER_NAME)):
+                self.web_session.HAWKULAR_PROVIDER_NAME)) and count < 10:
             self.web_driver.refresh()
+            count += 1
         assert WebDriverWait(self, 10).until(lambda s: not ui_utils(self.web_session).isElementPresent(By.XPATH,
                                                                                                        "//a[contains(@title,'Name: {}')]".format(
```

Comment:
Perhaps write a new ui_utils method that will wait for the element to be present - so that the code is reusable.